<a href="https://colab.research.google.com/github/kiplangat-david/Movie_land/blob/main/Movieland_sasakazi.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>


## **MOVIELAND**
# **Table of Contents**
- Introduction
- Objectives
- Description of the Data

- Importing Libraries

- Importing Data

- Data Cleaning

- Data Analysis
- Summary and Conclusion

# **Introduction**

Movieland is a production company conducting market research for their next project. Since Reviews provide insights into products, they would like to know from a given review dataset which movie was the most rated and what review was given by the viewer who liked it the best

# **Objectives**
1. Find out which movie had the highest rating 
2. Find out the number of reviewers who gave the highest rating
3. Find out the least rating and review detail 

# **Data Description**

Our data for this project was taken from Kaggle. This is the [link](https://www.kaggle.com/datasets/harshitshankhdhar/imdb-dataset-of-top-1000-movies-and-tv-shows?resource=download) to the dataset


# **Data Features**

-Our dataset has 13 features. Below are the descriptions of the features

1. **Poster_Link** - Link of the poster that imdb using

2. **Series_Title** - Name of the movie

3. **Released_Year** - Year at which that movie released

4. **Certificate** - Certificate earned by that movie

5. **Runtime** - Total runtime of the movie

6. **Genre** - Genre of the movie

7. **IMDB_Rating** - Rating of the movie at IMDB site

8. **Overview** - mini story/ summary

9. **Meta_score** - Score earned by the movie

10. **Director** - Name of the Director

11. **Star1,Star2,Star3,Star4** - Name of the Stars

12. **No_of_votes** - Total number of votes

13. **Gross** - Money earned by that movie


# **Importing Libraries**
Here, we will import pandas and numpy libraries which are useful for data anlysis and numerical calculations respectively.

In [3]:
#First let's import necessary libraries
import pandas as pd
import numpy as np

# **Loading Data**
Here, we will load our data then check the top, bottom and random sample records to see if our data rows and columns holds the expected correct data. 
We will also check the general information inorder for us to have a good understanding of our data

In [4]:
#load dataset
df = pd.read_csv('/content/imdb_top_1000.csv')

#Preview top 5 records
df.head()

Poster_Link  \
0  https://m.media-amazon.com/images/M/MV5BMDFkYT...   
1  https://m.media-amazon.com/images/M/MV5BM2MyNj...   
2  https://m.media-amazon.com/images/M/MV5BMTMxNT...   
3  https://m.media-amazon.com/images/M/MV5BMWMwMG...   
4  https://m.media-amazon.com/images/M/MV5BMWU4N2...   

               Series_Title Released_Year Certificate  Runtime  \
0  The Shawshank Redemption          1994           A  142 min   
1             The Godfather          1972           A  175 min   
2           The Dark Knight          2008          UA  152 min   
3    The Godfather: Part II          1974           A  202 min   
4              12 Angry Men          1957           U   96 min   

                  Genre  IMDB_Rating  \
0                 Drama          9.3   
1          Crime, Drama          9.2   
2  Action, Crime, Drama          9.0   
3          Crime, Drama          9.0   
4          Crime, Drama          9.0   

                                            Overview  Meta_score  \
0  Two imprisoned men bond over a number of years...        80.0   
1  An organized crime dynasty's aging patriarch t...       100.0   
2  When the menace known as the Joker wreaks havo...        84.0   
3  The early life and career of Vito Corleone in ...        90.0   
4  A jury holdout attempts to prevent a miscarria...        96.0   

               Director           Star1           Star2          Star3  \
0        Frank Darabont     Tim Robbins  Morgan Freeman     Bob Gunton   
1  Francis Ford Coppola   Marlon Brando       Al Pacino     James Caan   
2     Christopher Nolan  Christian Bale    Heath Ledger  Aaron Eckhart   
3  Francis Ford Coppola       Al Pacino  Robert De Niro  Robert Duvall   
4          Sidney Lumet     Henry Fonda     Lee J. Cobb  Martin Balsam   

            Star4  No_of_Votes        Gross  
0  William Sadler      2343110   28,341,469  
1    Diane Keaton      1620367  134,966,411  
2   Michael Caine      2303232  534,858,444  
3    Diane Keaton      1129952   57,300,000  
4    John Fiedler       689845    4,360,000

In [5]:
#preview bottom 5 records
df.tail()

Poster_Link  \
995  https://m.media-amazon.com/images/M/MV5BNGEwMT...   
996  https://m.media-amazon.com/images/M/MV5BODk3Yj...   
997  https://m.media-amazon.com/images/M/MV5BM2U3Yz...   
998  https://m.media-amazon.com/images/M/MV5BZTBmMj...   
999  https://m.media-amazon.com/images/M/MV5BMTY5OD...   

               Series_Title Released_Year Certificate  Runtime  \
995  Breakfast at Tiffany's          1961           A  115 min   
996                   Giant          1956           G  201 min   
997   From Here to Eternity          1953      Passed  118 min   
998                Lifeboat          1944         NaN   97 min   
999            The 39 Steps          1935         NaN   86 min   

                        Genre  IMDB_Rating  \
995    Comedy, Drama, Romance          7.6   
996            Drama, Western          7.6   
997       Drama, Romance, War          7.6   
998                Drama, War          7.6   
999  Crime, Mystery, Thriller          7.6   

                                              Overview  Meta_score  \
995  A young New York socialite becomes interested ...        76.0   
996  Sprawling epic covering the life of a Texas ca...        84.0   
997  In Hawaii in 1941, a private is cruelly punish...        85.0   
998  Several survivors of a torpedoed merchant ship...        78.0   
999  A man in London tries to help a counter-espion...        93.0   

             Director              Star1              Star2           Star3  \
995     Blake Edwards     Audrey Hepburn     George Peppard   Patricia Neal   
996    George Stevens   Elizabeth Taylor        Rock Hudson      James Dean   
997    Fred Zinnemann     Burt Lancaster   Montgomery Clift    Deborah Kerr   
998  Alfred Hitchcock  Tallulah Bankhead        John Hodiak   Walter Slezak   
999  Alfred Hitchcock       Robert Donat  Madeleine Carroll  Lucie Mannheim   

              Star4  No_of_Votes       Gross  
995     Buddy Ebsen       166544         NaN  
996   Carroll Baker        34075         NaN  
997      Donna Reed        43374  30,500,000  
998  William Bendix        26471         NaN  
999  Godfrey Tearle        51853         NaN

In [6]:
#Preview random 5 records
df.sample(5)

Poster_Link  \
468  https://m.media-amazon.com/images/M/MV5BMTQ4NT...   
634  https://m.media-amazon.com/images/M/MV5BOWE2MD...   
134  https://m.media-amazon.com/images/M/MV5BOGNlNm...   
508  https://m.media-amazon.com/images/M/MV5BYmVmMG...   
905  https://m.media-amazon.com/images/M/MV5BNDY2OT...   

                    Series_Title Released_Year Certificate  Runtime  \
468              I, Daniel Blake          2016          UA  100 min   
634                     Serenity          2005       PG-13  119 min   
134  Baahubali 2: The Conclusion          2017          UA  167 min   
508               Hable con ella          2002           R  112 min   
905                       Hævnen          2010           R  118 min   

                         Genre  IMDB_Rating  \
468                      Drama          7.9   
634  Action, Adventure, Sci-Fi          7.8   
134              Action, Drama          8.2   
508    Drama, Mystery, Romance          7.9   
905             Drama, Romance          7.6   

                                              Overview  Meta_score  \
468  After having suffered a heart-attack, a 59-yea...        78.0   
634  The crew of the ship Serenity try to evade an ...        74.0   
134  When Shiva, the son of Bahubali, learns about ...         NaN   
508  Two men share an odd friendship while they car...        86.0   
905  The lives of two Danish families cross each ot...        65.0   

            Director              Star1           Star2              Star3  \
468        Ken Loach       Laura Obiols      Dave Johns     Hayley Squires   
634      Joss Whedon     Nathan Fillion     Gina Torres   Chiwetel Ejiofor   
134   S.S. Rajamouli            Prabhas  Rana Daggubati     Anushka Shetty   
508  Pedro Almodóvar     Rosario Flores   Javier Cámara  Darío Grandinetti   
905     Susanne Bier  Mikael Persbrandt   Trine Dyrholm     Markus Rygaard   

                Star4  No_of_Votes       Gross  
468      Sharon Percy        53818     258,168  
634        Alan Tudyk       283310  25,514,517  
134  Tamannaah Bhatia        75348  20,186,659  
508    Leonor Watling       104691   9,284,265  
905       Wil Johnson        38491   1,008,098

Our dataset has expected rows and column records

In [7]:
#Check general information
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 16 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Poster_Link    1000 non-null   object 
 1   Series_Title   1000 non-null   object 
 2   Released_Year  1000 non-null   object 
 3   Certificate    899 non-null    object 
 4   Runtime        1000 non-null   object 
 5   Genre          1000 non-null   object 
 6   IMDB_Rating    1000 non-null   float64
 7   Overview       1000 non-null   object 
 8   Meta_score     843 non-null    float64
 9   Director       1000 non-null   object 
 10  Star1          1000 non-null   object 
 11  Star2          1000 non-null   object 
 12  Star3          1000 non-null   object 
 13  Star4          1000 non-null   object 
 14  No_of_Votes    1000 non-null   int64  
 15  Gross          831 non-null    object 
dtypes: float64(2), int64(1), object(13)
memory usage: 125.1+ KB


Our dataset has 1000 rows and 15 columns

In [8]:
df.columns

Index(['Poster_Link', 'Series_Title', 'Released_Year', 'Certificate',
       'Runtime', 'Genre', 'IMDB_Rating', 'Overview', 'Meta_score', 'Director',
       'Star1', 'Star2', 'Star3', 'Star4', 'No_of_Votes', 'Gross'],
      dtype='object')

In [9]:
df.dtypes

Poster_Link       object
Series_Title      object
Released_Year     object
Certificate       object
Runtime           object
Genre             object
IMDB_Rating      float64
Overview          object
Meta_score       float64
Director          object
Star1             object
Star2             object
Star3             object
Star4             object
No_of_Votes        int64
Gross             object
dtype: object

# **Data Cleaning**
Here, we will first select the columns which are of important to our analysis then proceed to clean those columns

In [10]:
#Select necessary columns
imdb = df[['Series_Title','Overview','IMDB_Rating','No_of_Votes']]

In [11]:
imdb.head()

Series_Title  \
0  The Shawshank Redemption   
1             The Godfather   
2           The Dark Knight   
3    The Godfather: Part II   
4              12 Angry Men   

                                            Overview  IMDB_Rating  No_of_Votes  
0  Two imprisoned men bond over a number of years...          9.3      2343110  
1  An organized crime dynasty's aging patriarch t...          9.2      1620367  
2  When the menace known as the Joker wreaks havo...          9.0      2303232  
3  The early life and career of Vito Corleone in ...          9.0      1129952  
4  A jury holdout attempts to prevent a miscarria...          9.0       689845

- We selected the above columns based on our desired outcomes. The information in these columns is sufficient to answer our questions.
Let's now proceed and clean these columns

In [12]:
#Check missing values
imdb.isnull().sum()

Series_Title    0
Overview        0
IMDB_Rating     0
No_of_Votes     0
dtype: int64

- There are no missing values.
Let's now proceed to data exploration

# **Data Analysis**
Here we will focus on answering our objectives

#Question 1.
# Which movie had the highest rating?

In [13]:
#Q1.
#Which movie had the highest rating?
top_rating = imdb[['Series_Title','IMDB_Rating']].sort_values(by = ['IMDB_Rating'], ascending=False)
top_rating.head(1)

Series_Title  IMDB_Rating
0  The Shawshank Redemption          9.3

The The Shawshank Redemption movie had a rating of 9.3 which was the highest

In [14]:
dept_gender_salary =  imdb.groupby(['Series_Title','IMDB_Rating'])['IMDB_Rating']

#Question 2
#How many reviewers gave it the highest rating?

In [15]:
#Q2
#How many reviewers gave it the highest rating?


#we will locate the movie with the highest rating and check the No_of_ Votes.
#The numbeer of vots here represent the viewers who rated the movie
reviewers = imdb[['Series_Title','IMDB_Rating','No_of_Votes']].sort_values(by = ['IMDB_Rating'], ascending=False)
reviewers.head(1)


#We can also locate the rating and the number of votes
reviewers.iloc[:,[1,2]].head(1)

IMDB_Rating  No_of_Votes
0          9.3      2343110

The most rated movie had 2343110 votes which for this case represent the viewers

#Question 3
#What is it's least rating and review details?

In [16]:
#Q3
#What is it's least rating and review details?
least_rate_review = imdb[['Overview','IMDB_Rating']].sort_values(by = ['IMDB_Rating'], ascending=True)
least_rate_review.head(1)


Overview  IMDB_Rating
999  A man in London tries to help a counter-espion...          7.6

- The least rated movie had a rating of 7.6 and the eview details were "A man in London tries to help a counter-espion..."	

# **Summary and Conclusion**

- The highest rated movie was The Shawshank Redemption with a rating of 9.3 and a total of 2343110 reviewers
- 7.6 was the least recorded rating. The movie of this rating was about a man in London who is trying to help an espion